spark on yarn problem -  column status valueDelegate setBytes too short

2016-03-08 08:29:13,954 ERROR io.hops.transaction.handler.RequestHandler: Rollback the TX
2016-03-08 08:29:13,954 ERROR io.hops.metadata.util.RMUtilities: HOP :: Error commiting finishRPC 
com.mysql.clusterj.ClusterJDatastoreException: For field status column status valueDelegate setBytes, error executing objectSetValue. Caused by com.mysql.clusterj.ClusterJUserException:Data length 1,025 too long. Column 'status' can only accept data of length 1,000.
	at com.mysql.clusterj.core.metadata.AbstractDomainFieldHandlerImpl.objectSetValue(AbstractDomainFieldHandlerImpl.java:289)
	at com.mysql.clusterj.tie.NdbRecordSmartValueHandlerImpl.set(NdbRecordSmartValueHandlerImpl.java:742)
	at com.mysql.clusterj.tie.NdbRecordSmartValueHandlerImpl.invoke(NdbRecordSmartValueHandlerImpl.java:769)
	at com.mysql.clusterj.core.metadata.$Proxy45.setStatus(Unknown Source)
	at io.hops.metadata.ndb.dalimpl.yarn.rmstatestore.CompletedContainersStatusClusterJ.createPersistable(CompletedContainersStatusClusterJ.java:113)
	at io.hops.metadata.ndb.dalimpl.yarn.rmstatestore.CompletedContainersStatusClusterJ.update(CompletedContainersStatusClusterJ.java:71)
	at io.hops.ha.common.TransactionStateImpl.persistAllocateResponsesToAdd(TransactionStateImpl.java:566)
	at io.hops.ha.common.TransactionStateImpl.persist(TransactionStateImpl.java:221)
	at io.hops.metadata.util.RMUtilities$33.performTask(RMUtilities.java:2228)
	at io.hops.transaction.handler.LightWeightRequestHandler.execute(LightWeightRequestHandler.java:50)
	at io.hops.transaction.handler.RequestHandler.handle(RequestHandler.java:52)
	at io.hops.transaction.handler.RequestHandler.handle(RequestHandler.java:47)
	at io.hops.metadata.util.RMUtilities.finishRPC(RMUtilities.java:2241)
	at io.hops.metadata.util.RMUtilities.finishRPCs(RMUtilities.java:2042)
	at io.hops.ha.common.TransactionStateImpl$RPCFinisher.run(TransactionStateImpl.java:857)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.mysql.clusterj.ClusterJUserException: Data length 1,025 too long. Column 'status' can only accept data of length 1,000.
	at com.mysql.clusterj.tie.NdbRecordImpl.setBytes(NdbRecordImpl.java:317)
	at com.mysql.clusterj.tie.NdbRecordOperationImpl.setBytes(NdbRecordOperationImpl.java:486)
	at com.mysql.clusterj.tie.NdbRecordOperationImpl.setBytes(NdbRecordOperationImpl.java:492)
	at com.mysql.clusterj.tie.NdbRecordSmartValueHandlerImpl.setBytes(NdbRecordSmartValueHandlerImpl.java:512)
	at com.mysql.clusterj.core.metadata.AbstractDomainFieldHandlerImpl$6.objectSetValue(AbstractDomainFieldHandlerImpl.java:858)
	at com.mysql.clusterj.core.metadata.AbstractDomainFieldHandlerImpl.objectSetValue(AbstractDomainFieldHandlerImpl.java:287)
	... 17 more
2016-03-08 08:29:14,004 INFO org.apache.hadoop.yarn.server.resourcemanager.scheduler.common.fica.FiCaSchedulerApp: Completed container: container_1457386138580_0009_01_000114 in state: COMPLETED event:FINISHED

--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/io/hops/ha/common/TransactionStateImpl.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/io/hops/ha/common/TransactionStateImpl.java
@@ -89,6 +89,7 @@ import java.util.concurrent.ConcurrentSkipListSet;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.LinkedBlockingQueue;
+import org.apache.commons.lang.StringUtils;
 import org.apache.hadoop.yarn.api.records.Container;
 import org.apache.hadoop.yarn.api.records.ContainerStatus;
 import org.apache.hadoop.yarn.api.records.impl.pb.ContainerPBImpl;
@@ -507,7 +508,13 @@ public class TransactionStateImpl extends TransactionState {
       
       Map<String, byte[]> completedContainersStatuses = new HashMap<String, byte[]>();
       for(ContainerStatus status: lastResponse.getCompletedContainersStatuses()){
-        completedContainersStatuses.put(status.getContainerId().toString(), ((ContainerStatusPBImpl)status).getProto().toByteArray());
+             ContainerStatus toPersist = status;
+        if(status.getDiagnostics().length()>1000){
+          toPersist = new ContainerStatusPBImpl(((ContainerStatusPBImpl)status).getProto());
+          toPersist.setDiagnostics(StringUtils.abbreviate(status.getDiagnostics(), 1000));
+        }
+        completedContainersStatuses.put(status.getContainerId().toString(), 
+                ((ContainerStatusPBImpl)toPersist).getProto().toByteArray());
       }
       
       AllocateResponsePBImpl toPersist = new AllocateResponsePBImpl();

